kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+            feeIncomeEntry.setTransactionLedgerEntrySequenceNumber(sequenceHelper.getSequenceCounter());
+    
+            feeIncomeEntry.setChartOfAccountsCode(feeIncomeChartCode);
+            feeIncomeEntry.setAccountNumber(feeIncomeAccountNumber);
+            feeIncomeEntry.setFinancialObjectCode(feeIncomeObjectCode);
+            feeIncomeEntry.setFinancialSubObjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankFinancialSubObjectCode());
+            feeIncomeEntry.setSubAccountNumber(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankSubAccountNumber());
+            feeIncomeEntry.setProjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankProjectCode());
+    
+            // retrieve object type
+            objectCode = getObjectCodeService().getByPrimaryIdForCurrentYear(feeIncomeChartCode, feeIncomeObjectCode);
+            if ( objectCode == null ) {
+                LOG.fatal("Specified income object code: " + feeIncomeChartCode + "-" + feeIncomeObjectCode + " does not exist - failed to generate foreign draft income entries", new RuntimeException() );
+                return false;
+            }
+            feeIncomeEntry.setFinancialObjectTypeCode(objectCode.getFinancialObjectTypeCode());       
+            feeIncomeEntry.setTransactionLedgerEntryAmount(feeAmount);
+            feeIncomeEntry.setTransactionDebitCreditCode(reverseEntries?GL_DEBIT_CODE:GL_CREDIT_CODE);
+            feeIncomeEntry.setFinancialBalanceTypeCode(KFSConstants.BALANCE_TYPE_ACTUAL);
+    
+            document.addPendingEntry(feeIncomeEntry);
+            sequenceHelper.increment();
+            
+            // create the offset entry
+            offsetEntry = new GeneralLedgerPendingEntry(feeIncomeEntry);
+            getGeneralLedgerPendingEntryService().populateOffsetGeneralLedgerPendingEntry(document.getPostingYear(), feeIncomeEntry, sequenceHelper, offsetEntry);
+    
+            document.addPendingEntry(offsetEntry);
+            sequenceHelper.increment();
+        }
+        return true;
+    }
+    
+    /**
+     * Adds up the amounts of all cash to offset GeneralLedgerPendingEntry records on the given AccountingDocument
+     * 
+     * Copied from the GL Pending entry service since that one does not make any distinction between
+     * expense and encumbrance balance types
+     * 
+     * @param glPostingDocument the accounting document total the offset to cash amount for
+     * @return the offset to cash amount, where debited values have been subtracted and credited values have been added
+     */
+    protected Map<String,KualiDecimal> getNonOffsetActualTotalsByChart(GeneralLedgerPostingDocument glPostingDocument) {
+        Map<String,KualiDecimal> totals = new HashMap<String, KualiDecimal>();
+        for (GeneralLedgerPendingEntry glpe : glPostingDocument.getGeneralLedgerPendingEntries()) {
+            if ( KFSConstants.BALANCE_TYPE_ACTUAL.equals(glpe.getFinancialBalanceTypeCode()) ) {
```

Comment:
This constant has been deprecated. Should be getting this from SystemOptions. See LaborBalanceSummaryReportServiceImpl.getActualBalanceTypes(Integer fiscalYear).